File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/adapter/api/FileOp.java
// FileOp.java

package com.bigboxcode.designpattern.adapter.api;

public class FileOp implements File{

    public String readFile() {
        // Code to read from file

        System.out.println("Reading from file");

        return "some dummy response read from file";
    }

    public void writeFile(String input) {
        // Write to file related code here

        System.out.println("Writing to file: " + input);
    }
}

File: design-patterns/go/adapter/file_api/file_op.go
// file_op.go

package main

import "fmt"

type FileOp struct {
}

func NewFileOp() (fileOp *FileOp) {
	fileOp = &FileOp{}
	return
}

func (fileOp *FileOp) ReadFile() (string) {
	// Code to read from file

	fmt.Println("Reading from file")
	return "some dummy response read from file"
}

func (fileOp *FileOp) WriteFile(input string) {
	// Write to file related code here

	fmt.Printf("Writing to file: %v\n", input)
}